src/core/thread/mesh_forwarder.cpp

```diff
@@ -796,6 +796,8 @@ Mac::TxFrame *MeshForwarder::HandleFrameRequest(Mac::TxFrames &aTxFrames)
 
     mSendBusy = true;
 
+    frame->SetTxTimestampEnabled(mSendMessage->IsTxTimestampEnabled());
```

Comment:
By doing this here, this won't work for data polls and CSL, right? I know this is for testing only but just letting you or our verification know that they shouldn't expect this wo work with SEDs.